File: API/src/routers/dirs.py
from fastapi import APIRouter, Request, Response, Depends
from src.schemas.response.dirs import *
from src.schemas.request.dirs import *
from src import services
from src.utils.validators import validate_auth

router = APIRouter(
    prefix="/dirs", 
    tags=["dirs"], 
    dependencies=[Depends(validate_auth)]
    )

@router.get("/list/{path:path}", response_model=ListDirsResponse)
async def list_dirs(path: str, request: Request) -> ListDirsResponse:
    return await services.list_dirs(path, request)

@router.get("/size/{path:path}", response_model=SizeDirResponse)
async def size_dir(path: str) -> SizeDirResponse:
    return await services.size_dir(path)

@router.post("/create", response_model=CreateDirResponse)
async def create_dir(path: str, request: Request) -> CreateDirResponse:
    return await services.create_dir(path, request)

@router.post("/rename", response_model=RenameDirResponse)
async def rename_dir(data: RenameDirRequest, request: Request) -> RenameDirResponse:
    return await services.rename_dir(data, request)

@router.post("/copy", response_model=CopyDirResponse)
async def copy_dir(data: CopyDirRequest, request: Request) -> CopyDirResponse:
    return await services.copy_dir(data, request)


@router.post("/add-favourite", response_model=AddFavouriteResponse)
async def add_fav(path: str, request: Request) -> AddFavouriteResponse:
    return await services.add_fav_dir(path, request)

@router.post("/rm-favourite", response_model=AddFavouriteResponse)
async def remove_fav(path: str, request: Request) -> AddFavouriteResponse:
    return await services.remove_fav_dir(path, request)

@router.delete("/delete", response_model=DeleteDirResponse)
async def delete_dir(path: str, request: Request) -> DeleteDirResponse:
    return await services.delete_dir(path, request)

File: WEB/src/utils/api/dirs.js
import axios from "axios";
import config from "../../../public/config.json";
import { getCookie } from "../utils.js";

const API_BASE = config.APIURL;

export const listDirs = async (path) =>
  axios.get(`${API_BASE}/dirs/list/${path}`, {
    withCredentials: true,
    headers: {
      "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
    },
  });

export const addFavDir = async (path) => {
  axios.post(
    `${API_BASE}/dirs/add-favourite?path=${path}`,
    {},
    {
      withCredentials: true,
      headers: {
        "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
      },
    }
  );
};

export const rmFavDir = async (path) => {
  await axios.post(
    `${API_BASE}/dirs/rm-favourite?path=${encodeURIComponent(path)}`,
    {},
    {
      withCredentials: true,
      headers: {
        "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
      },
    }
  );
};

export const createDir = async (path) => {
  return axios.post(
    `${API_BASE}/dirs/create?path=${encodeURIComponent(path)}`,
    {},
    {
      withCredentials: true,
      headers: {
        "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
      },
    }
  );
};

export const renameDir = (path, new_name) => {
  return axios.post(
    `${API_BASE}/dirs/rename`,
    { path: path, new_name: new_name },
    {
      withCredentials: true,
      headers: {
        "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
      },
    }
  );
};

export const deleteDir = async (path) => {
  return axios.delete(`${API_BASE}/dirs/delete`, {
    params: { path },
    withCredentials: true,
    headers: {
      "X-CSRF-TOKEN": getCookie("CSRF_ACCES_TOKEN"),
    },
  });
};
